### ISSUE TITLE
Secondary client state returned by `selectTreatmentAndStatus` does not update on timeout

### ISSUE THREAD

**Issue by WillsB3** [splitio/redux-client#113]
When calling `selectTreatmentAndStatus()` and passing a non-default key the returned state object contains an `hasTimedout` property, however this never gets set to `true` even after the (non-default) client times out.

To replicate this issue I:
1. Implemented a call to `selectTreatmentAndStatus()` and got the application working as expected.
2. Used Chrome DevTools network tab to request block the `sdk.split.io` domain.

When the request blocking is enabled my app just hangs on a loading spinner indefinitely (because i'm showing a spinner until `ready` is true here):

```js
const mapStateToProps = (state) => {
  const splitState = selectTreatmentAndStatus(
    state.splitio,
    FLAG_NAME,
    state.user._id
  )

  const isTimedOut = splitState.hasTimeout && !splitState.isReady

  return {
    …
    treatment: splitState.treatment,
    ready: splitState.isReady || isTimedOut,
  }
}

```

This is somewhat related to #112, and I've observed this behaviour while trying to work around that issue.

**Aside:** Why does the object returned by `selectTreatmentAndStatus's` not contain an `isTimedout` convenience field? Could it…? As it is at the moment I'm having to compute it (based on [these docs](https://help.split.io/hc/en-us/articles/360038825091-React-SDK#:~:text=Formally%2C%20isTimedout%20is%20equivalent%20to%20hasTimeout%20%26%26%20!isReady)) every time I need it.

**Comment by EmilianoSanchez** [comment]
Hi @WillsB3 ,

You are right. That behavior is an issue that we plan to address in an upcoming release, although there is no ETA yet.

Unlike the main SDK client (the one with the default key), the events of non-default clients do not dispatch actions (except when it implies an update of treatment evaluations, like SDK_READY, SDK_READY_FROM_CACHE, and SDK_UPDATE events). As a result, the Split’s piece of state isn’t updated, causing your component not to re-render, and `selectTreatmentAndStatus` is not called again to retrieve the updated client status. 

Besides fixing this issue, we'll include your suggestion regarding `isTimedout` field, which will make the API more consistent. Thank you for pointing that out!

**Comment by EmilianoSanchez** [comment]
Hi @WillsB3 ,

We have just released version 1.14.0 of the Redux SDK, which includes your suggestions among other improvements.

```
- Added `status` property to Split reducer's slice of state to track the SDK events of non-default clients (Related to https://github.com/splitio/redux-client/issues/113).
- Added `lastUpdate` and `isTimedout` properties to the object returned by the `getStatus` helper and `selectTreatmentAndStatus` and `selectTreatmentWithConfigAndStatus` selectors, to expose the last event timestamp and the timedout status of the SDK clients (Related to https://github.com/splitio/redux-client/issues/113).
 ```

I checked your code snippet example on my side, and it's working as expected now. 
I will close the issue for now, but please feel free to reopen it if you find that it's not properly resolved.

Thanks
